### PR TITLE
fix float32 for lm_head

### PR DESCRIPTION
**Pull request by pathfinder-pf**
## Motivation
for float32，accuracy decrease
the model output is different from tunix
image

## Modifications
add dtype for lm_head

## Accuracy Tests
- server
jax == 0.7.1 libtpu==0.0.20, flax==0.0.12, transformers==4.57.1
```
commit a314ef13940f6f20b7470c62638a8ebeb72c1303
export JAX_PLATFORMS=cpu
rm -rf /tmp/jit_cache
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache JAX_USE_SHARDY_PARTITIONER=0 python3 -u -m sgl_jax.launch_server \
--model-path deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B \
--trust-remote-code  \
--tp-size=1 \
--device=cpu \
--mem-fraction-static=0.05 \
--chunked-prefill-size=2048 \
--download-dir=/tmp \
--dtype=float32 \
--attention-backend=native \
--max-running-requests 64 \
--skip-server-warmup \
--page-size=64  \
--max-total-tokens=257536 \
--random-seed=27 \
--precompile-token-paddings=128 \
--precompile-bs-paddings=64 \
--disable-overlap-schedule \
--disable-precompile \
--use-sort-for-toppk-minp \
--enable-precision-tracer \
--log-requests \
--log-requests-level=3 
```
- start trace
curl -H "Content-Type: application/json" -X POST  http://localhost:30000/start_trace -d '{"req_num":1,"save_tensor":true}'
- compare the difference with tunix 
https://github.com/google/tunix/blob/main/tests/model_alignment/qwen_align_test.py

- the output's difference
image

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/models/qwen3_moe.py [hunk 1]
```diff
@@ -353,6 +353,8 @@ def __init__(
             self.lm_head = ParallelLMHead(
                 config.vocab_size,
                 config.hidden_size,
+                dtype=self.dtype,
```
Comment (aolemila) [inline python/sgl_jax/srt/models/qwen3_moe.py:356]: Is `self.dtype` configured by `--dtype`?

Comment (pathfinder-pf) [reply python/sgl_jax/srt/models/qwen3_moe.py:356]: yes
